<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>iPAS vignette • iPAS</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="iPAS vignette">
<meta property="og:description" content="iPAS">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iPAS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/iPAS_vignette.html">iPAS vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="iPAS_vignette_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>iPAS vignette</h1>
            
      
      
      <div class="hidden name"><code>iPAS_vignette.Rmd</code></div>

    </div>

    
    
<p>This is a simple introduction to the iPAS R package. This package performs pathway enrichment for differential expression data (RNA-seq, microarray, etc.). iPAS stands for integrative Pathway Activity Signatures, our method for constructing “signatures” of transcriptional response to signaling pathway perturbation using L1000 microarray data from the NIH LINCS project. The signatures are “integrative” in that they integrate cellular response across a diverse set of cell lines used in LINCS.</p>
<p>This vignette will show how the package works and how to perform simple pathway enrichment and create plots to analyze the output.</p>
<div id="loading-the-package" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-the-package" class="anchor"></a>Loading the package</h3>
</div>
<div id="loading-example-rna-seq-differntial-expression-data" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-example-rna-seq-differntial-expression-data" class="anchor"></a>Loading example RNA-seq differntial expression data</h3>
<p>First load the example RNA-seq data. The data originally came from differential expression signatures created from a gene silencing experiment. The genes have been shuffled as the data has not yet been published.</p>
<p>The format of the data is a matrix where the rownames are entrez IDs of the genes and the column names are the names of each sample or experiment. The numbers represent log2-transformed fold-change of counts per transcript (FPKM - Fragments Per Kilobase Million) from control to treated samples.</p>
<pre><code>##         example_1 example_2  example_3
## 1       0.0000000  0.000000 -0.0267561
## 503538  0.0472849 -0.148369  0.4159920
## 29974   0.0420521  0.381958  0.0000000
## 2       0.0316558 -3.229630  0.0000000
## 144571  0.1637790  0.000000  0.1874540
## 144568 -0.1971360  0.276755  0.0000000</code></pre>
</div>
<div id="pathway-enrichment-with-ipas" class="section level3">
<h3 class="hasAnchor">
<a href="#pathway-enrichment-with-ipas" class="anchor"></a>Pathway enrichment with iPAS</h3>
<p>The <em>iPAS_enrich</em> function performs pathway enrichment for each sample (this may take a few minutes). The function scores the differential expression from the input sample by its similarity to the iPAS signature for each pathway. By default, we use Pearson correlation for the similarity, but cosine similarity and the dot product between the two are given as options.</p>
<p>The significance of the pathway score is established via a permutation test. We permute the input signature <em>n</em> times (n=1000 by default) and calculate a null distribution of <em>n</em> pathway scores. We use the null distribution of scores to calculate an empirical p-value and a z-score.</p>
<p>Here we set the number of permutations to 100 to run the example quickly.</p>
</div>
<div id="review-results" class="section level3">
<h3 class="hasAnchor">
<a href="#review-results" class="anchor"></a>Review results</h3>
<p>You may view the pathway scores as a table.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span><span class="op">$</span><span class="va">example_1</span><span class="op">$</span><span class="va">ensemble</span></code></pre></div>
<pre><code>## # A tibble: 181 x 23
##    pathway cell_line  score z_score p_emp direction gene_scores query_scaled
##    &lt;chr&gt;   &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;     &lt;list&gt;      &lt;list&gt;      
##  1 hsa054… ensemble   0.911    2.26  0    Positive  &lt;NULL&gt;      &lt;NULL&gt;      
##  2 hsa030… ensemble  -1.08    -1.85  0.01 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
##  3 hsa049… ensemble  -0.513   -1.69  0.01 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
##  4 hsa046… ensemble   1.26     1.61  0.04 Positive  &lt;NULL&gt;      &lt;NULL&gt;      
##  5 hsa045… ensemble  -0.22    -1.68  0.05 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
##  6 hsa049… ensemble  -0.271   -1.63  0.06 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
##  7 hsa052… ensemble  -0.656   -1.53  0.06 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
##  8 hsa052… ensemble   1.05     1.58  0.07 Positive  &lt;NULL&gt;      &lt;NULL&gt;      
##  9 hsa047… ensemble  -0.378   -1.56  0.08 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
## 10 hsa047… ensemble  -0.707   -1.38  0.08 Negative  &lt;NULL&gt;      &lt;NULL&gt;      
## # … with 171 more rows, and 15 more variables: pas_scaled &lt;list&gt;,
## #   ens_weight &lt;dbl&gt;, score_type &lt;chr&gt;, overlap &lt;list&gt;, null_dist &lt;list&gt;,
## #   gene_scores_ens &lt;list&gt;, pathway_desc &lt;chr&gt;, size &lt;int&gt;, num_comps &lt;int&gt;,
## #   largest_comp &lt;dbl&gt;, path_nodes &lt;named list&gt;, path_nodes_list &lt;named list&gt;,
## #   Category_Lv1 &lt;chr&gt;, Category_Lv2 &lt;chr&gt;, Category_New &lt;chr&gt;</code></pre>
</div>
<div id="plot-results" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-results" class="anchor"></a>Plot results</h3>
<p>The <em>iPAS_bar</em> function creates a bar plot of the top scoring pathways for a sample. Here, bar height represents the empirical p-value (negative log10 transformed, so higher is more significant) and the color represents the z-score. The dotted line shows the cutoff for significance at a p&lt;0.05 level.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/iPAS_bar.html">iPAS_bar</a></span><span class="op">(</span><span class="va">res</span>, experiment <span class="op">=</span> <span class="fl">1</span>, num_top <span class="op">=</span> <span class="fl">15</span>, fill <span class="op">=</span> <span class="st">"z_score"</span>, bar_height <span class="op">=</span> <span class="st">"neg_log10_p"</span><span class="op">)</span></code></pre></div>
<p><img src="iPAS_vignette_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p>The <em>iPAS_heatmap</em> function creates a heatmap of pathway z-scores for each sample. The heatmap is clustered by rows (pathways) and columns (samples) so that a user can see which pathways score similarly across samples.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/iPAS_heatmap.html">iPAS_heatmap</a></span><span class="op">(</span><span class="va">res</span>, pathway_names <span class="op">=</span> <span class="st">"both"</span>, category <span class="op">=</span> <span class="st">"Signaling"</span><span class="op">)</span></code></pre></div>
<p><img src="iPAS_vignette_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>The <em>iPAS_density</em> function shows the null distribution of pathway scores (or z-scores) for one sample. The score or z-score of the sample is shown by a red line, along with a p-value.</p>
<p>The <em>iPAS_density_facet</em> function does the same for all samples at once.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/iPAS_density.html">iPAS_density</a></span><span class="op">(</span><span class="va">res</span>, path <span class="op">=</span> <span class="st">"hsa04150"</span>, experiment <span class="op">=</span> <span class="fl">1</span>, score_type <span class="op">=</span> <span class="st">"score"</span><span class="op">)</span></code></pre></div>
<p><img src="iPAS_vignette_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/iPAS_density_facet.html">iPAS_density_facet</a></span><span class="op">(</span><span class="va">res</span>, path <span class="op">=</span> <span class="st">"hsa04150"</span>, score_type <span class="op">=</span> <span class="st">"score"</span><span class="op">)</span></code></pre></div>
<p><img src="iPAS_vignette_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Nicholas Clark.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
